src/Core/WinSWCore/Configuration/YamlExtensionConfiguration.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using System.IO;
+using YamlDotNet.Serialization;
+
+namespace WinSW.Configuration
+{
+    public class YamlExtensionConfiguration
+    {
+        [YamlMember(Alias = "id")]
+        public string? ExtensionId { get; set; }
+
+        [YamlMember(Alias = "className")]
+        public string? ExtensionClassName { get; set; }
+
+        [YamlMember(Alias = "enabled")]
+        public bool Enabled { get; set; }
+
+        [YamlMember(Alias = "settings")]
+        public Dictionary<object, object>? Settings { get; set; }
+
+        public string GetId()
+        {
+            if (this.ExtensionId is null)
+            {
+                throw new InvalidDataException("Extension Id can't be empty");
```

Comment:
Not very helpful. Would it make sense  to keep the number of extension in the list?

Reply:
sorry. it's not clear to me the use case of having a number of extension??